File: molecular/src/test/java/com/hartwig/actin/molecular/orange/datamodel/virus/TestVirusFactory.java
package com.hartwig.actin.molecular.orange.datamodel.virus;

import org.apache.logging.log4j.util.Strings;
import org.jetbrains.annotations.NotNull;

public final class TestVirusFactory {

    private TestVirusFactory() {
    }

    @NotNull
    public static ImmutableVirusInterpreterEntry.Builder builder() {
        return ImmutableVirusInterpreterEntry.builder()
                .name(Strings.EMPTY)
                .integrations(0)
                .driverLikelihood(VirusDriverLikelihood.LOW);
    }
}

File: datamodel/src/test/kotlin/com/hartwig/actin/datamodel/molecular/driver/TestVirusFactory.kt
package com.hartwig.actin.datamodel.molecular.driver

import com.hartwig.actin.datamodel.molecular.evidence.TestClinicalEvidenceFactory
import com.hartwig.actin.datamodel.molecular.orange.driver.Virus
import com.hartwig.actin.datamodel.molecular.orange.driver.VirusType

object TestVirusFactory {

    fun createMinimal(): Virus {
        return Virus(
            isReportable = false,
            event = "",
            driverLikelihood = null,
            evidence = TestClinicalEvidenceFactory.createEmpty(),
            name = "",
            type = VirusType.OTHER,
            isReliable = false,
            integrations = 0
        )
    }
}
